Python/_warnings.c

```diff
@@ -1413,7 +1413,7 @@ _PyErr_WarnUnawaitedAgenMethod(PyAsyncGenObject *agen, PyObject *method)
 
 
 void
-_PyErr_WarnUnawaitedCoroutine(PyObject *coro)
+_PyErr_WarnUnawaitedCoroutine(PyGenObject *coro)
```

Comment:
This should only be called with a coroutine, so I think `PyCoroObject *` is correct type.
Maybe add an assert that `Py_TYPE(coro) == &PyCoro_Type`?